core/src/main/scala/org/apache/spark/rdd/CoalescedRDD.scala

```diff
@@ -78,6 +78,10 @@ private[spark] class CoalescedRDD[T: ClassTag](
     balanceSlack: Double = 0.10)
   extends RDD[T](prev.context, Nil) {  // Nil since we implement getDependencies
 
+  if (maxPartitions < 1) {
```

Comment:
Although I think you could use `require()` here, the change itself LGTM. I don't see a reason to allow repartitioning to 0 partitions.